dotnet/src/SemanticKernel.Core/Planning/KernelResultPlanExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// Extension methods for <see cref="KernelResult"/> with plan execution results.
+/// </summary>
+public static class KernelResultPlanExtensions
+{
+    /// <summary>
+    /// Returns plan execution results from <see cref="KernelResult"/>.
+    /// </summary>
+    /// <param name="kernelResult">Instance of <see cref="KernelResult"/>.</param>
+    public static IReadOnlyCollection<PlanResult> GetPlanResults(this KernelResult kernelResult)
+    {
+        var functionResult = kernelResult.FunctionResults.FirstOrDefault();
+
+        if (functionResult is not null && functionResult is PlanResult planResult)
+        {
+            return planResult.StepResults;
+        }
+
+        return Array.Empty<PlanResult>();
```

Comment:
Do we end up with two cases where an empty collection is returned i.e.

1. First function result is a PlanResult and has zero StepResults
2. First function is not a PlanResult

Should the caller be able to differentiate between these two cases? E.g. should we throw an exception if the KernelResult is no associated with a plan?

Reply:
Good catch! Although I think that instead of throwing the exception, we can simply return `null` if `KernelResult` is not associated with a plan. I pushed that change and updated XML documentation to describe this case. Thank you!